examples/linear_model/plot_elastic_net_precomputed_gram_matrix_with_weighted_samples.py

```diff
@@ -0,0 +1,52 @@
+"""
+==========================================================================
+Fitting an Elastic Net with a precomputed Gram Matrix and Weighted Samples
+==========================================================================
+
+The following example shows how to precompute the gram matrix
+while using weighted samples with an ElasticNet.
+
+If weighted samples are used, the design matrix must be centered and then
+rescaled by the sqare root of the weight vector before the gram matrix
+is computed.
+
+.. note::
+  `sample_weight`` vector is also rescaled to sum to `n_samples`, see comment
```

Comment:
I think that your comment shows a limitation of our documentation: we should not expect a user to look at the source code :)

Could you modify the documentation there as well: https://github.com/scikit-learn/scikit-learn/blob/42aff4e2e/sklearn/linear_model/_coordinate_descent.py#L731-L734

```
        sample_weight : float or array-like of shape (n_samples,), default=None
            Sample weight. Internally, the `sample_weight` vector will be rescaled to
            sum to `n_samples`.

            .. versionadded:: 0.23
```
Like this, we can directly redirect the user to the docstring of the documentation instead of the internal comment in the source code.